server/server/src/account_service.rs

```diff
@@ -153,31 +153,41 @@ pub enum GetUsageHelperError {
     UserNotFound,
 }
 
-pub fn get_usage<T>(
-    tree: &LazyTree<T>, sizes: &HashMap<Uuid, u64>,
+pub fn get_usage_helper<T>(
+    tree: &mut LazyTree<T>, sizes: &HashMap<Uuid, u64>,
 ) -> Result<Vec<FileUsage>, GetUsageHelperError>
 where
     T: TreeLike,
 {
-    Ok(tree
-        .ids()
+    let ids = tree.owned_ids();
+    let root = ids
         .iter()
-        .map(|&&file_id| {
-            let deleted = &tree
-                .implicit_deleted
-                .iter()
-                .filter(|&x| x.1 == &true)
-                .map(|x| *x.0)
-                .collect::<HashSet<Uuid>>();
-
-            let file_size = match deleted.contains(&file_id) {
+        .find(|file_id| match tree.find(file_id) {
+            Ok(f) => f.is_root(),
+            Err(_) => false,
+        })
+        .ok_or(GetUsageHelperError::UserNotFound)?;
+
+    let result = ids
```

Comment:
I think you're better off relying on `db.owned_files` for a set of ids this person owns

Reply:
that causes some headache with the borrow checker. owned files are borrowed by the tree. passing the tree and db.owned_files will clash. 
What's wrong with using the tree to check if a file is owned by the user or not?

Reply:
you can clone an array of ids before you create the trees, it's sort of the single place we're trying to maintain the idea of what files you're responsible for paying for. When that model evolves it'll be easier to change the code if there aren't multiple places we're expressing the same idea in slightly different ways.

It also eliminates finding the root O(n) and scanning the files, unwrapping and ultimately allocating your own info.

My gut says readability and performance win if you just clone the set of ids out of the db for a user